server/src/main/java/org/opensearch/indices/TermsLookup.java

```diff
@@ -104,14 +125,12 @@ public void writeTo(StreamOutput out) throws IOException {
         if (out.getVersion().onOrAfter(Version.V_2_17_0)) {
             out.writeBoolean(store);
         }
-    }
-
-    public String index() {
-        return index;
-    }
-
-    public String id() {
-        return id;
+        if (query != null) {
+            out.writeBoolean(true);
```

Comment:
I dont think you should write your own bool to ser/deser the query param
The current optional writeable implementation does that for you. 


```
 public <T extends Writeable> void writeOptionalWriteable(final Writer<T> writer, @Nullable T writeable) throws IOException {
        if (writeable != null) {
            writeBoolean(true);
            writer.write(this, writeable);
        } else {
            writeBoolean(false);
        }
    }